doc/reference/configuration/configuration_reference.rst

```diff
@@ -3,13 +3,280 @@
+
+.. confval:: replicasets.<replicaset_name>.leader
+
+    A replica set leader.
+    This option is in effect when ``manual`` :ref:`replication.failover <configuration_reference_replication_failover>` is used.
+
+    **Example**
+
+    ..  literalinclude:: /code_snippets/snippets/replication/instances.enabled/manual_leader/config.yaml
+        :language: yaml
+        :start-at: replicaset001:
+        :end-at: listen: 127.0.0.1:3303
+        :dedent:
+
+
+.. _configuration_reference_replicasets_name_bootstrap_leader:
+
+.. confval:: replicasets.<replicaset_name>.bootstrap_leader
+
+    A bootstrap leader for a replica set.
+    To specify a bootstrap leader manually, you need to set :ref:`replication.bootstrap_strategy <configuration_reference_replication_bootstrap_strategy>` to ``config``.
+
+    **Example**
+
+    ..  literalinclude:: /code_snippets/snippets/replication/instances.enabled/bootstrap_strategy/config.yaml
+        :language: yaml
+        :start-at: replicaset001
+        :end-at: listen: 127.0.0.1:3303
+        :dedent:
+
+
+.. _configuration_reference_replicasets_name_instances:
+
+.. confval:: replicasets.<replicaset_name>.instances
+
+    Instances that belong to this replica set. See :ref:`instances <configuration_reference_instances>`.
+
+
+.. _configuration_reference_replicasets_name_config_parameter:
+
+.. confval:: replicasets.<replicaset_name>.<config_parameter>
+
+    Any configuration parameter that can be defined in the replica set :ref:`scope <configuration_scopes>`.
+    For example, :ref:`iproto <configuration_reference_iproto>` and :ref:`database <configuration_reference_database>` configuration parameters defined at the replica set level are applied to all instances in this replica set.
+
+
+
+..  _configuration_reference_instances:
+
+instances
+*********
+
+.. NOTE::
+
+    ``instances`` can be defined in the replica set :ref:`scope <configuration_scopes>` only.
+
+-   :ref:`instances.\<instance_name\> <configuration_reference_instances_name>`
+-   :ref:`instances.\<instance_name\>.\<config_parameter\> <configuration_reference_instances_name_config_parameter>`
+
+.. _configuration_reference_instances_name:
+
+.. confval:: instances.<instance_name>
+
+    An instance name.
+
+
+.. _configuration_reference_instances_name_config_parameter:
+
+.. confval:: instances.<instance_name>.<config_parameter>
+
+    Any configuration parameter that can be defined in the instance :ref:`scope <configuration_scopes>`.
+    For example, :ref:`iproto <configuration_reference_iproto>` and :ref:`database <configuration_reference_database>` configuration parameters defined at the instance level are applied to this instance only.
+
+
+..  _configuration_reference_credentials:
+
+credentials
+-----------
+
+..  TODO: https://github.com/tarantool/doc/issues/3666
+
+.. NOTE::
+
+    ``credentials`` can be defined in any :ref:`scope <configuration_scopes>`.
+
+
+..  _configuration_reference_credentials_role:
+
+roles
+~~~~~
```

Comment:
It looks a bit confusing on the website:

* credentials
  * roles
    * credentials.roles
    * credentials.roles.<...>

I mean that we use just `roles` on the second level, but a fully qualified name at the lower level.

And it is inconsistent with options of the `config` section. It looks so:

* config
  * config.reload (not just reload!)
  * config.version
  * config.etcd.*

Reply:
I would also add that without some indentation non-qualified names are confusing for me. I'm scrolling here and there and sometimes it is hard to spot, whether, say, the given title is about top level option `roles` or about sub-section `roles` in the `credentials` section. Some difference in title font sizes helps, but only after some practice.

I guess that it is not an easy task to restyle the website in order to better render hierarchical documents. So I think that using fully qualified names would be a good choice.

Reply:
My bad :) Please see the current staging here:

https://docs.d.tarantool.io/en/doc/3.0-config-replication/reference/configuration/configuration_reference/#credentials